crates/network-primitives/src/traits.rs

```diff
@@ -63,10 +64,24 @@ pub trait ReceiptResponse {
 
 /// Transaction JSON-RPC response.
 pub trait TransactionResponse {
```

Comment:
Is this now just `Transaction` + additional functions to get the block context?

Reply:
Yep, although I'm not entirely across the context. It looks similar to how `ReceiptResponse` is a projection of the fields on a `TransactionReceipt`. Happy to be guided by owners on this